schemas/JSON/configuration/configuration.schema.0.1.json

```
@@ -0,0 +1,204 @@
+                        }
+                    }
+                },
+                "directives": {
+                    "type": "object", 
+                    "properties": {
+                        "requiredVersion": {
+                            "$ref": "#/defs/version"
+                        },
+                        "timeoutSeconds": {
+                            "type": "integer",
+                            "minimum": 0,
+                            "maximum": 6000,
+                            "description": "The number of seconds to wait before timing out."
+                        },
+                        "runAs": {
+                            "type": "string",
+                            "enum": [
+                              "admin",
+                              "user"
+                            ],
+                            "description": "Enum determining whether to run as admin or user. Default is user."
+                        },
+                        "sha256Hash": {
+                            "type": "string",
+                            "pattern": "^[A-Fa-f0-9]{64}$",
+                            "description": "SHA256 hash of the resource."
+                        },
+                        "optional": {
+                            "type": [ "boolean", "null" ],
+                            "description": "Indicates whether to inform the orchestrator to prompt user for install. Default is false."
+                        },
+                        "confirm": {
+                            "type": [ "boolean", "null" ],
+                            "description": "Indicates whether to prompt the user for confirmation before initiation."
+                        },
+                        "signerThumbprint": {
+                            "type": "string",
+                            "pattern": "^[A-Fa-f0-9]{40}$",
+                            "description": "The signer thumbprint."
+                        },
+                        "onFailure": {
+                            "type": "string",
+                            "enum": [
+                              "continue",
+                              "stop"
+                            ],
+                            "description": "Determines the behavior if a failure is encountered."
+                        }
+                    }
+                }
+            },
+            "required": [
+                "resourceName",
+                "id"
+            ]
+        },
+        "id": {
+            "type": "string",
+            "maxLength": 128,
+            "description": "The identifier of the resource."
+        },
+        "module": {
+            "type": "string",
+            "maxLength": 128,
+            "description": "The name of the module."
+        },
+        "description": {
+            "type": "string",
+            "maxLength": 512,
+            "description": "The description of the resource."

```

```suggestion
            "description": "The description of the desired state."
```
The point is to describe this invocation of the resource, not the resource itself.